pkg/lightning/mydump/loader.go

```diff
@@ -54,6 +54,9 @@ type SourceFileMeta struct {
 }
 
 func (m *MDTableMeta) GetSchema(ctx context.Context, store storage.ExternalStorage) (string, error) {
+	if m.SchemaFile.FileMeta.Path == "" {
+		return "", errors.Errorf("schema file for `%s`.`%s` not found", m.DB, m.Name)
```

Comment:
This is an improvement. However with `-no-schema` I don't expect it to read the schema file.

It looks like it wants to read the schema file if the table doesn't exist even with `-no-schema`. Here the problem is not that the schema file doesn't exist, the problem is that the table is missing.

Reply:
It sounds more reasonable. I'm going to close this PR and help merge your fix.